aspnetcore/performance/performance-best-practices/samples/3.0/Startup.cs

```diff
@@ -0,0 +1,110 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using performance_best_practices.Controllers;
+using System;
+using System.Threading.Tasks;
+
+namespace performance_best_practices
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddScoped<SearchService>();
+
+            services.AddControllers();
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseHttpsRedirection();
+
+            app.UseRouting();
+
+            app.UseAuthorization();
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapControllers();
+            });
+        }
+    }
+
+    public class SearchService : IDisposable {
```

Comment:
What's this for?

Reply:
> What's this for?
Auto code generation by VS to get the code to compile. It offered a fix and I clicked "OK". Please suggest better code.

Reply:
Remove it all and just implement the interface 😄 .

Reply:
`public class SearchService : IDisposable {`

> What's this for?

VS generated code so your snippet would compile.